ClemBot.Bot/bot/cogs/translate_cog.py

```diff
@@ -1,272 +1,275 @@
+        if not input_lang:
+           await self.error_handling(ctx, input_lang)
+           return
+        elif not output_lang:
+           await self.error_handling(ctx, output_lang)
+           return
+
+        text = ' '.join(input[2:])
+
+        params = {
+            'api-version': '3.0',
+            'from': input_lang,
+            'to': output_lang
+        }
+
+        body = [{
+            'text': text
+        }]
+
+        async with await session.post(url=TRANSLATE_API_URL, params=params, headers=headers, json=body) as resp:
+            response = json.loads(await resp.text())
+
+        embed = discord.Embed(title='Translate', color=Colors.ClemsonOrange)
+        name = 'Translated to ' + LANGUAGE_SHORT_CODE_TO_NAME[response[0]['translations'][0]['to']]
+        embed.add_field(name=name, value=response[0]['translations'][0]['text'], inline=False)
+        await ctx.send(embed=embed)
+
+    async def error_handling(self, ctx, error: str):
+
+        if not error:   #Double making sure that the string isn't actually set so this doesn't get published multiple times.
+              await self.bot.messenger.publish(Events.on_set_pageable_text,
+                                                embed_name='Languages',
+                                                field_title='Language entry is not valid. Here are the available languages:',
+                                                pages=get_language_list(self),
+                                                author=ctx.author,
+                                                channel=ctx.channel)
+        else:
+             return
+
+        
+
+
+
+    async def translate_given_lang(self, ctx, language: str, input):
+
+        output_lang = language
+        text = ' '.join(input[1:])
+
+        params = {
+            'api-version': '3.0',
+            'to': output_lang
+        }
+
+        body = [{
+            'text': text
+        }]
+        
+        async with await session.post(url=TRANSLATE_API_URL, params=params, headers=headers, json=body) as resp:
+            response = json.loads(await resp.text())
+
+        embed = discord.Embed(title='Translate', color=Colors.ClemsonOrange)
+        name = 'Translated to ' + LANGUAGE_SHORT_CODE_TO_NAME[response[0]['detectedLanguage']['language']]
+        embed.add_field(name='Confidence Level:', value=response[0]['detectedLanguage']['score'], inline=True)
+        embed.add_field(name=name, value=response[0]['translations'][0]['text'], inline=False)
+        await ctx.send(embed=embed)
+        return
+
+
+
+async def get_lang_code(self, ctx, language: str):
+    
+    language_lower = language.lower()
+    if language_lower in LOWER_CODE_TO_CODE:
+        return LOWER_CODE_TO_CODE[language_lower]
+    elif language_lower in LOWER_LANGUAGE_TO_CODE:
+        return LOWER_LANGUAGE_TO_CODE[language_lower]
+    else:
+            return None
+
+
+def get_language_list(self):

```

Comment:
A function taking a `self` parameter is bad. It signifies that it should be a method of a class, not a function. Plus, it's only used when calling the `chunk_list` function which doesn't use it at all.